vortex-array/src/data.rs

```diff
@@ -65,11 +76,12 @@ impl ArrayData {
         self.buffer
     }
 
-    pub fn child(&self, index: usize, dtype: &DType) -> Option<&Array> {
+    pub fn child(&self, index: usize, dtype: &DType, len: usize) -> Option<&Array> {
```

Comment:
both this and the dtype check feel potentially unnecessary, or at least maybe could be debug assertions?